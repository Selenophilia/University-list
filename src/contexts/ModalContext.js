import React, { useReducer, createContext } from "react";
import PropTypes from "prop-types";
import { SHOW_MODAL, CLOSE_MODAL, CHANGE_MODAL_TITLE } from "./constants";

const initialState = {
  content: () => {},
  visible: false,
  title: ""
};

const ModalContext = createContext({ state: initialState });

const modalReducer = (state = initialState, action) => {
  switch (action.type) {
    case SHOW_MODAL:
      return {
        ...state,
        visible: true,
        content: action.payload.content,
        title: action.payload.title
      };
    case CLOSE_MODAL:
      return {
        ...state,
        visible: false,
        title: "",
        content: () => {}
      };
    case CHANGE_MODAL_TITLE:
      return {
        ...state,
        title: action.payload
      };
    default:
      return state;
  }
};

export const ModalProvider = ({ children }) => {
  const [state, dispatch] = useReducer(modalReducer, initialState);

  const showModal = (payload) => {
    dispatch({
      type: SHOW_MODAL,
      payload
    });
  };

  const changeModalTitle = (title) => {
    dispatch({
      type: CHANGE_MODAL_TITLE,
      payload: title
    });
  };

  const closeModal = () => {
    dispatch({ type: CLOSE_MODAL });
  };

  return (
    <ModalContext.Provider
      value={{ state, showModal, closeModal, changeModalTitle }}
    >
      {children}
    </ModalContext.Provider>
  );
};

ModalProvider.propTypes = {
  children: PropTypes.node.isRequired
};

export default ModalContext;
