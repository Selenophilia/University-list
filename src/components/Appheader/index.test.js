import React from "react";
import { render, fireEvent, screen } from "@testing-library/react";
import { ApolloProvider } from "@apollo/client";
import client from "../../utils/client";
import ModalContext from "../../contexts/ModalContext";
import Navbar from ".";

const showModal = jest.fn();
const renderRoot = (props) => {
  return render(
    <ApolloProvider client={client}>
      <ModalContext.Provider value={{ showModal }}>
        <Navbar {...props} />
      </ModalContext.Provider>
    </ApolloProvider>
  );
};

it("renders navbar correctly ", () => {
  const { asFragment } = renderRoot();
  expect(asFragment()).toMatchSnapshot();
});

it("calls showModal if sign in button is clicked", () => {
  renderRoot();
  fireEvent.click(screen.getByTestId("signIn"));
  expect(showModal).toHaveBeenCalledTimes(1);
});

it("calls showModal if log in button is clicked", () => {
  renderRoot();
  fireEvent.click(screen.getByTestId("login"));
  expect(showModal).toHaveBeenCalledTimes(1);
});
