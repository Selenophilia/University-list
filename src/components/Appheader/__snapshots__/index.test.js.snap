// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders navbar correctly  1`] = `
<DocumentFragment>
  <div
    class="makeStyles-container-1"
  >
    <header
      class="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary makeStyles-header-2 MuiPaper-elevation4"
    >
      <div
        class="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
      >
        <h5
          class="MuiTypography-root makeStyles-title-3 MuiTypography-h5"
        >
          <div
            class="MuiBox-root MuiBox-root-11 makeStyles-user-5"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root makeStyles-icon-6"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M5 13.18v4L12 21l7-3.82v-4L12 17l-7-3.82zM12 3L1 9l11 6 9-4.91V17h2V9L12 3z"
              />
            </svg>
            University Listings
          </div>
        </h5>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-login-8"
          data-testid="login"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            Log in
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-signin-7"
          data-testid="signIn"
          label="Sign up"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            Sign up
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </header>
  </div>
</DocumentFragment>
`;
